cromulent/model.py

```diff
@@ -913,6 +928,129 @@ def _toJSON(self, done, top=None):
 		else:
 			return d
 
+	def _toJSON_fast(self, done, top=None):
+		"""Serialize as JSON."""
+		# If we're already in the graph, return our URI only
+		# This should only be called from the factory!
+
+		# This should ONLY be used in Python3.6+
+		# as it relies on a stable order to a regular dict
+		# which was not the case before 3.6
+
+		# id, type, _label is the default.
+		if not self._factory.id_type_label and id(self) in done:
+			return self.id
+
+		# Can't pass in self as a param
+		if top is None:
+			top = self
+
+		# Add back context at the top, if set
+		result = {}
+		if top is self and not id(self) in done and self._factory.context_uri: 
+			result['@context'] = self._factory.context_uri
+
+		if self.id:
+			# Could be a bnode
+			result['id'] = self.id
+		if self.type:
+			# Could be an external reference
+			result['type'] = self.type
+		try:
+			result['_label'] = self._label
+		except:
+			pass
+
+		# Need only minimal representation of self
+		if (self._factory.id_type_label and id(self) in done) or (top is not self and not self._embed):
+			# limit to only id, type, label
+			return result
+		else:	
+			# otherwise, we're about to serialize the resource completely
+			done[id(self)] = 1			
+
+		d = self.__dict__.copy()
+		del d['_factory']
+		del d['id']
+
+		# Need to do in order now to get done correctly ordered
+		kvs = list(d.items())
+		if self._factory.order_json:
+			KOH = self._factory.key_order_hash
+			kodflt = self._factory.key_order_default
+			# in place sort is 2% faster than sorted
+			# rather than over an iterable
+			# and sorting is 10% of srlz cost
+			# and use setdefault to amortize cost of default
+			# over multiple calls to the factory's hash
+			kvs.sort(key=lambda x: KOH.setdefault(x[0], kodflt))
+			
+		# tbd vs done is to ensure that in a DAG rather than a tree
+		# that it is breadth first, not depth first.
+		# This doesn't catch the pattern A-B-C-D / A-E-D,
+		# (D will be under C, not under E) 
+		# Reimplemented this in a single depth-first loop
+		# and it was less than 2% faster (!)
+		# This is just not that expensive
+
+		tbd = []
+		for (k, v) in kvs:
+			k = self._property_name_map.get(k, k)
+			if isinstance(v, ExternalResource):
+				if self._factory.linked_art_boundaries and \
+					not self._linked_art_boundary_okay(top, k, v):
+					# never follow, so just add to done
+					done[id(v)] = 1
+				else:
+					tbd.append(id(v))
+			elif type(v) is list:
+				for ni in v:
+					if isinstance(ni, ExternalResource):
+						if self._factory.linked_art_boundaries and \
+							not self._linked_art_boundary_okay(top, k, ni):
+							# never follow, so just add to done
+							done[id(ni)] = 1							
+						else:
+							tbd.append(id(ni))
+				# For completeness should check list-of-datetime here too
+			elif isinstance(v, datetime.datetime):
+				# replace with string
+				kvs[k] = v.strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
I don't think this is right. Line 977 above has `kvs` set to a list, but `k` here is going to be a `str` key. It's going to result in a `TypeError` on encountering a `datetime` value.

Reply:
True!

Reply:
Moved to second iteration.